preprocessy/pipelines/_base.py

```diff
@@ -143,6 +170,24 @@ def __insert(self, index, func, params):
 
     def add(self, func=None, params=None, **kwargs):
 
+        """Function to add another function to the pipeline after it has been constructed
```

Comment:
Add which positions function can be added. Add what happens when index, before, after all are provided. (priority = index > after > before)